src/snmalloc/backend/strictprovenancebackend.h

```diff
@@ -0,0 +1,241 @@
+#if defined(SNMALLOC_CHECK_CLIENT) && !defined(OPEN_ENCLAVE)
+/**
+ * Protect meta data blocks by allocating separate from chunks for
+ * user allocations. This involves leaving gaps in address space.
+ * This is less efficient, so should only be applied for the checked
+ * build.
+ *
+ * On Open Enclave the address space is limited, so we disable this
+ * feature.
+ */
+#  define SNMALLOC_META_PROTECTED
+#endif
+
+namespace snmalloc
+{
+  /**
+   * This class implements the standard backend for handling allocations.
+   * It abstracts page table management and address space management.
+   */
+  template<SNMALLOC_CONCEPT(ConceptPAL) PAL>
+  class StrictProvenanceBackend : public CommonConfig
+  {
+  public:
+    using Pal = PAL;
+    class SlabMetadata : public FrontendSlabMetadata
+    {
+      friend StrictProvenanceBackend;
+
+    private:
+      /**
+       * In the strict provenance world we need to retain a pointer to the
+       * entire slab (actually bounded to mmapped region) for various uses:
+       *  - on deallocation, to clear the slab and allow merging in backend
+       *  - on CHERI+MTE as the authority for reversioning
+       * This is a high privilege pointer and we should consider protecting it
+       * (perhaps by sealing).
+       */
+      capptr::Chunk<void> chunk_ptr;
+    };
+    using PageMapEntry = FrontendMetaEntry<SlabMetadata>;
+
+  private:
+    using ConcretePagemap =
+      FlatPagemap<MIN_CHUNK_BITS, PageMapEntry, PAL, false>;
+
+  public:
+    using Pagemap = BasicPagemap<
+      StrictProvenanceBackend,
+      PAL,
+      ConcretePagemap,
+      PageMapEntry,
+      false>;
+
+    static constexpr bool CONSOLIDATE_PAL_ALLOCS = false;
```

Comment:
There are lots of missing doc comments here, but this one in particular is confusing.  Is it actually meant to be public or is it used only in the definition of the ranges below?